### PR TITLE
Add more 9x19mm guns

### PR DESCRIPTION
**Pull request by Rail-Runner**
#### Summary

SUMMARY: Content "Add more 9x19mm guns"
#### Purpose of change

Adding more variety to obscure guns that can be found.
#### Describe the solution

Add entries for more 9x19mm guns, their magazines, add them to item groups and Generic Guns migration lists. Also added an item group for extremely rare guns in general (that would be too rare to get even a frequency of 1 in obscure gun lists; )
#### Describe alternatives you've considered

So as to not add guns that are actually nearly useless, I decided to not add guns which use very rare ammo in addition to being rare themselves.
#### Testing

Launched the game to make sure that no errors happen; loaded, unloaded and fired newly-added guns to make sure that they behave correctly.
#### Additional context

Suggestions related to actual probability of guns spawning in item groups are welcome.

### CODE REVIEW

**Comment by Rail-Runner** [comment]
> Also, one of the guns you have appears to have no folding stock, but is described as such?

Thing is, the folding stock gunmod doesn't seem to work like it should; it reduces volume for some reason, but should probably reduce just length instead (with reduction being individual for every gun). So I decided to assume length to be length of a folded gun, and just give SLOW_WIELD flag to those guns to represent having to unfold the stock before use.

**Comment by tenmillimaster** [comment]
The numbers I gave you were raw gunbroker results, not direct numbers to use in the item groups. You need to either: Adjust the entire group so gunbroker results and each gun are 1:1, or adjust the raw results given so that the results:indices ratios matches gunbroker:indices ratios.


For example: There were 324 Luger P08's given. I just searched for completed makarov pistol listings: 267 listings. 
The ratio in game needs to be 324 P08 : 267 Makarovs, roughly 1.2:1.

Therefore you can either multiply _every_ item in `guns_pistol_obscure` by (GB hits makarov)/(current weight makarov), or 2.67 and then insert your 324 for P08's directly into the distribution...

OR  

Insert 121 (the calculated equivalent in game weight, as determined below).
(GB hits P08)/(GB hits makarov) * (current weight makarov)=
324/267*100
121.3

**Comment by Rail-Runner** [comment]
Mostly fixed now, though I didn't have the exact numbers for PPSh-41 or STEN SMGs, to assign spawn frequency to MP 40 based on that.

File: data/json/items/gun/9mm.json [hunk 1]
```diff
@@ -322,6 +322,241 @@
...
+  {
+    "id": "mauser_m714",
+    "copy-from": "mauser_c96",
+    "type": "GUN",
+    "name": { "str": "detachable-magazine Mauser" },
+    "description": "This is a variant of the Mauser C96 pistol chambered for 9x19mm, which uses a detachable magazine rather than an integral one.",
+    "weight": "1070 g",
+    "volume": "615 ml",
+    "price": 90000,
+    "pocket_data": [
```
Comment (ZhilkinSerg) [inline data/json/items/gun/9mm.json:476]:
```
ERROR : src/item_factory.cpp:1437 [void Item_factory::check_definitions() const] warnings for type mauser_c96:

cannot have more than one pocket that handles ammo (MAGAZINE or MAGAZINE_WELL)
```

File: data/json/items/gun/9mm.json [hunk 2]
```diff
@@ -322,6 +322,241 @@
...
+      [ "mechanism", 4 ],
+      [ "muzzle", 1 ],
+      [ "rail mount", 1 ],
+      [ "sights mount", 1 ],
+      [ "stock", 1 ],
+      [ "underbarrel mount", 1 ]
+    ],
+    "flags": [ "RELOAD_ONE" ],
+    "pocket_data": [
+      { "pocket_type": "MAGAZINE", "ammo_restriction": { "9mm": 10 } },
```
Comment (Wishbringer) [inline data/json/items/gun/9mm.json:457]:
```suggestion
```

**Comment by Rail-Runner** [comment]
Should be working now.

**Pull request merged**